src/native/corehost/comhost/comhost.cpp

```diff
@@ -42,14 +43,45 @@ struct com_activation_context
     void **class_factory_dest;
 };
 
-using com_delegate_fn = int(STDMETHODCALLTYPE*)(com_activation_context*);
+#define ISOLATED_CONTEXT (void*)-1
+using com_delegate_fn = int(STDMETHODCALLTYPE*)(com_activation_context*, void*);
+using com_delegate_no_load_context_fn = int(STDMETHODCALLTYPE*)(com_activation_context*);
 
 namespace
 {
-    int get_com_delegate(hostfxr_delegate_type del_type, pal::string_t *app_path, com_delegate_fn *delegate)
+    int get_com_delegate_no_load_context(hostfxr_delegate_type del_type, get_function_pointer_fn get_function_pointer, void **delegate)
     {
-        return load_fxr_and_get_delegate(
-            del_type,
+        const pal::char_t* method_name;
+        switch (del_type)
+        {
+        case hostfxr_delegate_type::hdt_com_activation:
+            method_name = _X("GetClassFactoryForTypeInternal");
+            break;
+        case hostfxr_delegate_type::hdt_com_register:
+            method_name = _X("RegisterClassForTypeInternal");
+            break;
+        case hostfxr_delegate_type::hdt_com_unregister:
+            method_name = _X("UnregisterClassForTypeInternal");
+            break;
+        default:
+            return StatusCode::InvalidArgFailure;
+        }
+
+        return get_function_pointer(
+            _X("Internal.Runtime.InteropServices.ComActivator, System.Private.CoreLib"),
+            method_name,
+            UNMANAGEDCALLERSONLY_METHOD,
+            nullptr, // load context
+            nullptr, // reserved
+            delegate);
+    }
+
+    int get_com_delegate(hostfxr_delegate_type del_type, pal::string_t *app_path, com_delegate_fn *delegate, bool* del_has_load_context, void **load_context)
```

Comment:
I would rather we used `void** delegate` to avoid giving it a wrong type in half the cases. It looks really weird below to cast one function pointer to another.

Reply:
Or maybe as an alternative - take a pointer/ref to a small struct which stores both delegates and use that to have the function pointer always strongly typed.

I probably like this approach better as it avoids duplicating the casting code in the three call sites. The struct could also carry the boolean which says which one it is - or we could rely on one being null always as the indication instead.

Reply:
Wow, yeah - this is pretty weird. Guess who clearly shoe-horned in the fallback after an initial implementation?